tests/unittests/graphOptzTest.cpp

```diff
@@ -1922,3 +1922,29 @@ TEST_F(GraphOptz, ReshapePrivateVarOneUse) {
   // The new Variable should have the same shape as the original second Reshape.
   EXPECT_TRUE(V->getType()->dims().equals(reshape2));
 }
+
+TEST_F(GraphOptz, ConvertPlaceholdersToConstants) {
+  auto *input1 = mod_.createPlaceholder(ElemKind::FloatTy, {1}, "input1", true);
+  auto *input2 = mod_.createPlaceholder(ElemKind::FloatTy, {1}, "input2", true);
+  auto *input3 = mod_.createPlaceholder(ElemKind::FloatTy, {1}, "input3", true);
+  auto *save1 = F_->createSave(ctx_, "save1", input1);
+  auto *save2 = F_->createSave(ctx_, "save2", input2);
+  auto *save3 = F_->createSave(ctx_, "save3", input3);
+
+  // No variables, six PHs (3 inputs, 3 saves).
+  EXPECT_EQ(mod_.getVars().size(), 0);
+  EXPECT_EQ(mod_.getPlaceholders().size(), 6);
+
+  // Allocate two of the three inputs, but mark input2 of them as non-constant.
+  ctx_.allocate(input1);
+  ctx_.allocate(input2);
+  ::glow::convertPlaceholdersToConstants(mod_, ctx_, {input2});
```

Comment:
Why is "input3" not allocated here?

Reply:
I think that the intension is to verify that unallocated placeholders don't crash the procedure. It would probably be a good idea to add a comment.

Reply:
The way I was reading this was that unallocated placeholders are just not considered to be constants, as they have no tensor backing them. 

Do we have some intended usage here for when placeholders should be allocated? It seems we have three options -- (1) they are all allocated before calling `compile()`, (2) they are all allocated after calling `compile()`, or (3) they can be allocated before or after `compile()`. The blacklist is necessary for (1) and (3). I was reading this test case as more like (3).

Reply:
@jfix71 Yeah, the testcase is intended to exercise all the options; (1) allocated tensor/placeholder replaced by constant, (2) unallocated placeholder not replaced, and (3) allocated but blacklisted.